### DISCUSSION TITLE
Is it possibile to add an analytical term of energy to DeepMD models?

### DISCUSSION THREAD

**Discussion by omaraek** [deepmodeling/deepmd-kit#2242] [General]
Hi all, 

When I try to mix a different potential with other force fields I get the following error:
"Must not use GPU neighbor lists with hybrid pair style"  
which seems to be a problem of lammps. 
So I was wondering how complicated is it to implement an additive analytical energy term to the NN potentials (like LJ or Grimme's Dispersion and so on) within the frozen model of deepmd?

Thanks in advance 

Omar

**Comment by njzjz** [comment]
For LJ or Grimme, you can look at [deepmodeling/DMFF](https://github.com/deepmodeling/DMFF). It uses JAX to implement these terms. TensorFlow has a similar API as JAX,  so I think it should be okay. Or an interface to DMFF can be considered, as JAX provides [a tool](https://github.com/google/jax/blob/main/jax/experimental/jax2tf/jax2tf.py) to convert JAX functions to TF functions.
cc @WangXinyan940 @KuangYu 

Just curious, is there a reason to use GPU neighbor lists?

**Reply by omaraek** [reply]
Dear Jinzhe, 
Thank you for your suggestion, I will give a look at DMFF and JAX. 
On my cluster, the calculations slow down when I build the neighbor lists on CPUs. I use "-pk gpu 0 neigh no" to switch to CPU neighbor lists. Does this sound strange to you?
